### PR TITLE
A fluid-type post

### PR DESCRIPTION
**Pull request by mirisuzanne**
## Related Issue(s)
_Reminder to add related issue(s), if available._

- #530 

## Steps to test/reproduce
_Please explain how to best reproduce the issue and/or test the changes locally (including the pages/URLs/views/states to review)._

## Show me
_Provide screenshots/animated gifs/videos if necessary._

### CODE REVIEW

File: content/blog/2025/fluid-type.md [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+of course I have to stop using that preference.
+
+That leads us to an assumption that
+the browser preference will always be `16px`.
+We compare that to our mockups with `24px` fonts,
+do some basic arithmetic,
+and set our base font size to `1.5em`.
+_Good enough_.
+
+TL;DR never do pixel math with `em` and `rem` units.
```
Comment (dvdherron) [inline content/blog/2025/fluid-type.md:159]:
```suggestion
TL;DR: never do pixel math with `em` and `rem` units.
```

File: content/blog/2025/fluid-type.md [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+  /* no size is the best size */
+}
+```
+
+## Why should type be responsive at all?
+
+But I still have an issue
+because _the browser_ doesn't give me many options
+for setting the preference.
+And (not surprising)
```
Comment (dvdherron) [inline content/blog/2025/fluid-type.md:199]:
```suggestion
And (not surprisingly)
```

File: content/blog/2025/fluid-type.md [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+the common default `16px` font size
+feels fine to me on a phone,
+with limited line length,
+but it's smaller than I want
+when I have 20 inches to fill.
+
+Of course,
+I could set different defaults
+for different devices,
+But that doesn't help when I resize windows.
```
Comment (dvdherron) [inline content/blog/2025/fluid-type.md:218]:
```suggestion
but that doesn't help when I resize windows.
```

File: content/blog/2025/fluid-type.md [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+Even with the browser constantly running fullscreen,
+may laptop is only sometimes plugged into
+an external monitor.
+when I unplug it, I have less space available.
```
Comment (dvdherron) [inline content/blog/2025/fluid-type.md:222]:
```suggestion
When I unplug it, I have less space available.
```

File: content/blog/2025/fluid-type.md [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+I think browsers like Arc are on the right track
+with per-site style adjustments.
+
+We could take my responsive-default approach even farther
+by providing site-specific controls
+as part of our UI.
+We still want to make sure the defaults are reasonable,
+based on the global preferences,
+and only provide the UI as a progressive enhancement.
+To quote accessibility expert Kate Kalcevich…
```
Comment (dvdherron) [inline content/blog/2025/fluid-type.md:327]:
```suggestion
To quote accessibility expert Kate Kalcevich:
```

**Review by dvdherron — COMMENTED** [review]
Some copy feedback on a first pass.

**Review by dvdherron — APPROVED** [review]

File: content/blog/2025/fluid-type.md [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+generally with a `clamp()` function
+to integrate both `em`/`rem` and `vi`/`cqi` _inputs_
+in a responsive algorithm
+with lower and upper boundaries.
+See [utopia.fyi](https://utopia.fyi)
+for the most common approach:
+
+```css
+:root {
+  font-size: clamp(1.125rem, 1.0815rem + 0.2174vi, 1.25rem)
```
Comment (stacyk) [inline content/blog/2025/fluid-type.md:64]:
```suggestion
  font-size: clamp(1.125rem, 1.0815rem + 0.2174vi, 1.25rem);
```

File: content/blog/2025/fluid-type.md [hunk 7]
```diff
@@ -0,0 +1,356 @@
...
+It seems like Chromium browsers
+no longer allow setting a specific font size.
+Instead you can choose from the default `medium`
+or a list of larger and smaller options:
+
+- Very small
+- Small
+- Medium (recommended)
+- Large
+- Very large
```
Comment (stacyk) [inline content/blog/2025/fluid-type.md:112]: It does show the size when you go into the Customize Fonts panel and focus or hover over the Font Size slider
font-size-slider

File: content/blog/2025/fluid-type.md [hunk 8]
```diff
@@ -0,0 +1,356 @@
...
+  'name': "Kate Kalcevich"
+}) }}
+
+I mocked up a (stand-alone) custom element
+with a rough proof of concept.
+I'm sure there are ways to improve it
+such as adjusting the values,
+adding a form submit option,
+or tracking the preference in local storage --
+but I'm curious what you think.
```
Comment (stacyk) [inline content/blog/2025/fluid-type.md:340]: Should we add a link here so they know where to let us know what they think?

File: content/blog/2025/fluid-type.md (hunk 7 above)
Comment (mirisuzanne) [reply content/blog/2025/fluid-type.md:112]: Ah, that's interesting. I guess I hadn't gone deeper.

File: content/blog/2025/fluid-type.md [hunk 9]
```diff
@@ -0,0 +1,356 @@
...
+
+But I've already designed my site
+to have large text _relative to_ the user default.
+Ignoring the fluid bits for now,
+I set the root `font-size` to roughly `1.5em` --
+half again the size of the user setting.
+When my preference was set to the default `16px`,
+my site would already display text at `24px`.
+Now that I've said I _want_ `24px` font-size,
+my site has increased to `36px`!
```
[the remark this answers is not in the document]
Comment (stacyk) [reply content/blog/2025/fluid-type.md:101]: I don't think it matters that isn't what you are using as you could change that at any time after publishing this.

File: content/blog/2025/fluid-type.md [hunk 10]
```diff
@@ -0,0 +1,356 @@
...
+Even with the browser constantly running fullscreen,
+may laptop is only sometimes plugged into
+an external monitor.
+When I unplug it, I have less space available.
+
+Again I've heard a common refrain that
+"users don't resize their windows" --
+but I find that entirely unbelievable.
+In fact, I'd bet I'm in a very small minority
+by using mostly fullscreen apps.
```
[the remark this answers is not in the document]
Comment (mirisuzanne) [reply content/blog/2025/fluid-type.md:228]: We could link to a mastodon or bluesky survey, but we'd be getting response from other developers. I guess the framing could be something like:

Do you ever resize your browser window, aside from testing responsive designs as a developer?

- Yes (I'm a front-end dev)
- Yes (Not a front-end dev)
- No (I'm a front-end dev)
- No (Not a front-end dev)

[the remark this answers is not in the document]
Comment (stacyk) [reply content/blog/2025/fluid-type.md:228]: I think if you emphasize the "**aside from responsive testing**" then you might not need the "I'm a FED/not a FED" but that's my two cents.

[the remark this answers is not in the document]
Comment (mirisuzanne) [reply content/blog/2025/fluid-type.md:228]: It looks like Bluesky doesn't have polls - but I set this one on Mastodon to run for three days: https://front-end.social/@mia/113992409837989769

**Comment by mirisuzanne** [comment]
@stacyk I think I addressed your comments. @jgerigmeyer do you want to take a look as well? Ideally I'll post this today, since I'm out of the office tomorrow.

File: content/blog/2025/fluid-type.md [hunk 11]
```diff
@@ -0,0 +1,404 @@
...
+  I think we can improve on the common approaches.
+---
+
+{% import "quotes.macros.njk" as quotes %}
+{% import 'embed.macros.njk' as embed %}
+
+{% callout 'note' %}
+We talked about some of these ideas
+in our [Winging It](/2025/01/24/winging-it-15/) live stream
+last month,
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:33]:
```suggestion
last month
```

File: content/blog/2025/fluid-type.md [hunk 12]
```diff
@@ -0,0 +1,404 @@
...
+with Alan Stearns --
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:34]:
```suggestion
with Alan Stearns,
```

File: content/blog/2025/fluid-type.md [hunk 13]
```diff
@@ -0,0 +1,404 @@
...
+  font-size: clamp(1.125rem, 1.0815rem + 0.2174vi, 1.25rem);
+}
+```
+
+Those numbers look a bit magical,
+because they are.
+Utopia asks us to start from a range of font sizes
+defined in `px` values,
+and then it does a conversion to `rem`
+by assuming that `1rem === 16px`.
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:80]: This `===` feels very JavaScripty to me.

```suggestion
by assuming that `1rem == 16px`.
```

File: content/blog/2025/fluid-type.md [hunk 14]
```diff
@@ -0,0 +1,404 @@
...
+As long as that assumption hold true,
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:81]:
```suggestion
As long as that assumption holds true,
```

File: content/blog/2025/fluid-type.md [hunk 15]
```diff
@@ -0,0 +1,404 @@
...
+regardless of the existing design.
+I really do want sites to _usually_
+just give me text around `24px` (or 'large'),
+because that's _a pretty good default for me_.
+Sites with smaller body text
+would ideally increase their font size,
+but sites with the same size or larger text
+certainly shouldn't get _even bigger_.
+
+It's not just my site,
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:178]:
```suggestion
It's not just my site --
```

File: content/blog/2025/fluid-type.md [hunk 16]
```diff
@@ -0,0 +1,404 @@
...
+this has become a common practice across the industry.
+And it means some of my favorite sites
+already using large type
+become _too large_ when I set my browser preference.
+So I had to remove that preference.
+
+I see this situation play out over and over
+on the web.
+And then the lesson we often learn is
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:187]:
```suggestion
The lesson we often learn is
```

File: content/blog/2025/fluid-type.md [hunk 17]
```diff
@@ -0,0 +1,404 @@
...
+of course I have to stop using that preference.
+
+That leads us to an assumption that
+the browser preference will always be `16px`.
+We compare that to our mockups with `24px` fonts,
+do some basic arithmetic,
+and set our base font size to `1.5em`.
+_Good enough_.
+
+TL;DR: never do pixel math with `em` and `rem` units.
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:201]: Elsewhere in OddSite we're doing:

```suggestion
TL;DR -- Never do pixel math with `em` and `rem` units.
```

File: content/blog/2025/fluid-type.md [hunk 18]
```diff
@@ -0,0 +1,404 @@
...
+That's where we went wrong,
+by assuming that `16px === 1em` is a reliable fact.
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:203]:
```suggestion
by assuming that `16px == 1em` is a reliable fact.
```

File: content/blog/2025/fluid-type.md [hunk 19]
```diff
@@ -0,0 +1,404 @@
...
+
+```css
+html {
+  /* no size is the best size */
+}
+```
+
+## Why should type be responsive at all?
+
+But I still have an issue
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:238]:
```suggestion
But I still have an issue,
```

File: content/blog/2025/fluid-type.md [hunk 20]
```diff
@@ -0,0 +1,404 @@
...
+with limited line length,
+but it's smaller than I want
+when I have 20 inches to fill.
+
+Of course,
+I could set different defaults
+for different devices,
+but that doesn't help when I resize windows.
+Even with the browser constantly running fullscreen,
+may laptop is only sometimes plugged into
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:262]:
```suggestion
my laptop is only sometimes plugged into
```

File: content/blog/2025/fluid-type.md [hunk 21]
```diff
@@ -0,0 +1,404 @@
...
+at some target value in our designer mind,
+and the clamps are not related to my desired range of outcomes.
+Instead we're adding a slight responsiveness to the user value,
+and using the clamps to keep our fluid value
+within range of the user's intent.
+
+Relative units like `em` and `rem`
+are excellent for _internal relationships_ --
+keeping the scale consistent between
+body text and headings for example.
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:340]:
```suggestion
body text and headings, for example.
```

File: content/blog/2025/fluid-type.md [hunk 22]
```diff
@@ -0,0 +1,404 @@
...
+where my web-surfing preference for large text
+does not apply equally to all sites.
+When I'm reading a blog post,
+I just want a nice big view of the text.
+When I'm scanning bank transactions,
+I'm happy for something a bit more compact.
+I think browsers like Arc are on the right track
+with per-site style adjustments.
+
+We could take my responsive-default approach even farther
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:365]:
```suggestion
We could take my responsive-default approach even further
```

File: content/blog/2025/fluid-type.md [hunk 23]
```diff
@@ -0,0 +1,404 @@
...
+as part of our UI.
+We still want to make sure the defaults are reasonable,
+based on the global preferences,
+and only provide the UI as a progressive enhancement.
+To quote accessibility expert Kate Kalcevich:
+
+{{ quotes.blockquote({
+  'text': "The real advance in accessibility is providing options and adapting to user preferences.",
+  'name': "Kate Kalcevich"
+}) }}
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:376]: Can we link to a source for this quote? Not a big deal.

File: content/blog/2025/fluid-type.md [hunk 24]
```diff
@@ -0,0 +1,404 @@
...
+
+I mocked up a (stand-alone) custom element
+with a rough proof of concept.
+I'm sure there are ways to improve it
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:380]:
```suggestion
I'm sure there are ways to improve it,
```

File: content/blog/2025/fluid-type.md [hunk 25]
```diff
@@ -0,0 +1,404 @@
...
+but I'm curious what you think.
+
+{% import 'embed.macros.njk' as embed %}
+
+{{ embed.codepen(
+  id='yyLLErw',
+  title='Fluid-Type Custom Element',
+  user='miriamsuzanne',
+  height=600
+) }}
```
Comment (jgerigmeyer) [inline content/blog/2025/fluid-type.md:393]: This codepen has two misspellings of "calculate"/"calculator" as "calcuate"/"calcuator".

**Review by jgerigmeyer — APPROVED** [review]

File: content/blog/2025/fluid-type.md (hunk 23 above)
Comment (mirisuzanne) [reply content/blog/2025/fluid-type.md:376]: It was a talk that I saw in person, so not sure if there's a link

Comment (mirisuzanne) [reply content/blog/2025/fluid-type.md]: Found it, link added

File: content/blog/2025/fluid-type.md (hunk 25 above)
Comment (mirisuzanne) [reply content/blog/2025/fluid-type.md:395]: This should be fixed
